src/main/java/org/squiddev/plethora/integration/roost/IntegrationRoost.java

```diff
@@ -0,0 +1,62 @@
+package org.squiddev.plethora.integration.roost;
+
+import com.google.common.collect.Maps;
+import com.timwoodcreates.roost.Roost;
+import com.timwoodcreates.roost.data.DataChicken;
+import com.timwoodcreates.roost.data.DataChickenModded;
+import com.timwoodcreates.roost.data.DataChickenVanilla;
+import net.minecraft.entity.Entity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import org.squiddev.plethora.api.Injects;
+import org.squiddev.plethora.api.converter.ConstantConverter;
+import org.squiddev.plethora.api.converter.DynamicConverter;
+import org.squiddev.plethora.api.meta.IMetaProvider;
+import org.squiddev.plethora.api.meta.NamespacedMetaProvider;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Injects(Roost.MODID)
+public final class IntegrationRoost {
+	private IntegrationRoost() {
+	}
+
+	public static final ConstantConverter<ItemStack, DataChicken> GET_DATA_CHICKEN_FROM_ITEM = DataChicken::getDataFromStack;
+
+	public static final DynamicConverter<Entity, DataChicken> GET_DATA_CHICKEN_FROM_ENTITY = DataChicken::getDataFromEntity;
+
+	public static final IMetaProvider<DataChicken> META_DATA_CHICKEN = new NamespacedMetaProvider<>("roost", context -> {
+		Map<Object, Object> out = new HashMap<>();
+		DataChicken chicken = context.getTarget();
+		out.put("name", chicken.getName());
+
+		if (chicken instanceof DataChickenModded) {
```

Comment:
So you could technically split this into three meta providers (`DataChicken`, and then `DataChicken{Vanilla,Modded}`). It'll probably be uglier, but means we can do examples for each case.